go/vt/vtgate/executor.go

```diff
@@ -856,24 +855,18 @@ func (e *Executor) handleShow(ctx context.Context, safeSession *SafeSession, sql
 			RowsAffected: 2,
 		}, nil
 	case "create table":
-		if destKeyspace == "" && show.HasTable() {
-			// For "show create table", if there isn't a targeted keyspace already
-			// we can either get a keyspace from the statement or potentially from
-			// the vschema.
-
-			if !show.Table.Qualifier.IsEmpty() {
-				// Explicit keyspace was passed. Use that for targeting but remove from the query itself.
-				destKeyspace = show.Table.Qualifier.String()
-				show.Table.Qualifier = sqlparser.NewTableIdent("")
-				sql = sqlparser.String(show)
-			} else {
-				// No keyspace was indicated. Try to find one using the vschema.
-				tbl, err := e.VSchema().FindTable("", show.Table.Name.String())
-				if err == nil {
-					destKeyspace = tbl.Keyspace.Name
-				}
+		if !show.Table.Qualifier.IsEmpty() {
+			// Explicit keyspace was passed. Use that for targeting but remove from the query itself.
+			destKeyspace = show.Table.Qualifier.String()
+			show.Table.Qualifier = sqlparser.NewTableIdent("")
+		} else {
+			// No keyspace was indicated. Try to find one using the vschema.
+			tbl, err := e.VSchema().FindTable("", show.Table.Name.String())
```

Comment:
In this case, `FindTable` should take `destKeyspace` as input. If it was non-empty, the function will resolve the table against that specific keyspace. If empty, it will do a global search.

We should also change the error handling logic to return the error returned by `FindTable`. This will require us to change the `unknown_table` test case.

The err == nil logic is good. If a table was found in the vschema, then we should send it to that keyspace.

Something like
```
if err != nil {
  return nil, err
}
destKeyspace = tbl.Keyspace.Name
```